plutus-core/exe/Main.hs

```diff
@@ -636,8 +648,35 @@ runTypecheck (TypecheckOptions inp fmt) = do
 
 ---------------- Evaluation ----------------
 
+-- Convert a time in picoseconds into a readble format with appropriate units
+formatTime :: Double -> String
+formatTime t
+    | t >= 1e12 = printf "%.3f s"  (t/1e12)
+    | t >= 1e9  = printf "%.3f ms" (t/1e9)
+    | t >= 1e6  = printf "%.3f μs" (t/1e6)
+    | t >= 1e3  = printf "%.3f ns" (t/1e3)
+    | otherwise = printf "%f ps"   t
+
+{- Apply an evaluator to a program a number of times and report the mean execution
+time.  The first measurement is often significantly larger than the rest
+(perhaps due to warm-up effects), and this can distort the mean.  To avoid this
+we measure the evaluation time (n+1) times and discard the first result. -}
+timeEval :: NFData a => Integer -> (t -> a) -> t -> IO ()
+timeEval n evaluate prog =
+    if n <= 0 then error "Error: the number of repetitions should be at least 1"
```

Comment:
I'm not sure that that helped.  Having the guards is a bit clearer, but then emacs indented it like this:

```
    | otherwise = do
  (results, times) <- unzip . tail <$> for (replicate (fromIntegral (n+1)) prog) (timeOnce evaluate)
  let mean = (fromIntegral $ sum times) / (fromIntegral n) :: Double
```
It looks a bit odd with the stuff inside the do-block to the left of the guard.  I can cope with that though.